### PR TITLE
Use groups

### PR DESCRIPTION
**Pull request by danieleongari**
Switched to the new convention of using groups (e.g., `curated-cof_17162N3_v1`) to link the resulst.
TODO list:
- [x]  Update README
- [x] Test with v2

### CODE REVIEW

File: detail/graph.py [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
 """
 
-def get_aiida_link(cof_label, link_label):
+def get_aiida_link(group_node, extra_tag):
     import os
     from aiida.orm.querybuilder import QueryBuilder
-    from aiida.orm import  Node, StructureData, WorkFunctionNode
+    from aiida.orm import  Node, Group
 
     explore_url = os.getenv('EXPLORE_URL', "https://dev-www.materialscloud.org/explore/curated-cofs")
```
Comment (ltalirz) [inline detail/graph.py]: maybe move out of function (don't need to do this every time)

File: detail/graph.py [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
 
     qb = QueryBuilder()
-    qb.append(Node, filters={'label': cof_label}, tag='cof')
-    qb.append(WorkFunctionNode, filters={'attributes.function_name': {'==': 'link_outputs'}}, with_incoming='cof', tag='link')
-    qb.append(Node, edge_filters={'label': link_label}, with_outgoing='link')
+    qb.append(Group, filters={'uuid': group_node.uuid}, tag='group')
+    qb.append(Node, filters={'extras.curated-cof_tag': extra_tag}, with_group='group')
```
Comment (ltalirz) [inline detail/graph.py]: This is still one query per node (of course, they will be already faster than before). 
How about you do the following in `main.ipynb`:

 1. load the group (you already do)
 2. 
```
node_dict = {}
for node in self.group.nodes:
   node_dict[node.get_extra('curated-cof_tag')] = node
self.node_dict = node_dict
```
 3. use the node_dict without having to query

File: figure/query.py [hunk 3]
```diff
@@ -18,21 +18,26 @@
...
     for inp in inp_list:
         if inp in  ['henry_coefficient_average_co2', 'henry_coefficient_average_n2']:
             proj = 'henry_coefficient_average' #fix to distinguish co2/n2
         else:
             proj = inp
-        qb.append(Dict, project=['attributes.{}'.format(proj)], edge_filters={'label': get_link[inp]}, with_outgoing='link')
-    qb.append(Node, project=['label'], edge_filters={'label': 'orig_cif'}, with_outgoing='link')
+        qb.append(Dict, project=['attributes.{}'.format(proj)], filters={'extras.curated-cof_tag': get_tag[inp]},
+            with_group='curated-cof')
+    print(qb.all())
```
Comment (ltalirz) [inline figure/query.py]: leftover print

**Review by ltalirz — COMMENTED** [review]
Thanks - some suggestions + I'll come to ask about the "figure" plot

**Comment by danieleongari** [comment]
This works with `export_v012.aiida` (which means v0+v1+v2) I made on November 8th.
Please merge and test it.

**Comment by ltalirz** [comment]
merging now in order to test on mcloud

**Pull request merged**